Directory.Packages.props

```diff
@@ -109,4 +102,29 @@
     <PackageVersion Include="Xunit.Combinatorial" Version="$(XunitCombinatorialVersion)" />
     <PackageVersion Include="xunit.console" Version="$(XUnitVersion)"/>
   </ItemGroup>
+
+  <!-- Use different versions of Microsoft.Build.* depending on whether we are building in source-only
+       modes or normal modes. In normal modes we use MicrosoftBuildMinimumVersion, otherwise
+       we use MicrosoftBuildVersion. Note: This is not abstracted in Versions.props because source-only
+       modes will import a version override file after Versions.props. -->
+  <ItemGroup Condition="'$(DotNetBuildSourceOnly)' == 'true'">
```

Comment:
@rainersigwald I think this maintains the same split versions between source build and not as before but I figure someone from msbuild should double check that.

Reply:
@dotnet/kitten ? I can look at some point but am out sick.

Reply:
Yeah, intention was to maintain the split versions. When building in MSFT mode, it should maintain the same behavior as the repo in isolation.

Reply:
The key bit is that I changed the way the split versions are maintained so that it works with non-SB VMR builds.